### PR TITLE
ARC-1449 redis reconnect

### PR DESCRIPTION
**Pull request by joshkay10**
**What's in this PR?**
When we hit a readonly error, its most likely that we are connected to the read-only replica.

In this case we ask ioredis to reconnect to the redis instance and it should point back to master on reconnect!!

**Why**
During failover testing we had an incident when we didn't reconnect to the master instance and kept trying to write to the replica which is readonly 😢 

**Affected issues**  
ARC-1449

**How has this been tested?**  
Testing in stage with a micros failover event for REDIS

I also was able to force my local docker to create a slave instance and connect to that move it to read only and see the logs keep attempting to reconnect until i restored the master.

**Whats Next?**
Nothing!! hopefully

### CODE REVIEW

File: src/config/redis-info.ts [hunk 1]
```diff
@@ -6,7 +6,14 @@ export const getRedisInfo = (connectionName: string): IORedis.RedisOptions => ({
...
 	db: 0,
+	reconnectOnError(err) {
+		const targetError = "READONLY";
 
+		if (err.message.includes(targetError)) {
+			return 1;
+		}
+		return false;
+	},
 	// TODO find out if we still need these options
```
Comment (joshkay10) [inline src/config/redis-info.ts:17]: They put a fix in on that github link, but people are still complaining, will investigate further outside this PR

File: src/config/redis-info.ts [hunk 2]
```diff
@@ -6,7 +6,14 @@ export const getRedisInfo = (connectionName: string): IORedis.RedisOptions => ({
 	port: Number(envVars.REDISX_CACHE_PORT) || 6379,
 	host: envVars.REDISX_CACHE_HOST || "127.0.0.1",
 	db: 0,
+	reconnectOnError(err) {
+		const targetError = "READONLY";
```
Comment (joshkay10) [inline src/config/redis-info.ts:10]: considered making `const targetErrors = ["READONLY", "BLAH BLAH"];` work, but the reconnection logic that is handled by `ioredis` was working for the examples that I could find.... If we run into problems in the future will be pretty easy to slap it in.

File: src/config/redis-info.ts [hunk 3]
```diff
@@ -6,7 +6,14 @@ export const getRedisInfo = (connectionName: string): IORedis.RedisOptions => ({
 	port: Number(envVars.REDISX_CACHE_PORT) || 6379,
 	host: envVars.REDISX_CACHE_HOST || "127.0.0.1",
 	db: 0,
+	reconnectOnError(err) {
```
Comment (joshkay10) [inline src/config/redis-info.ts:9]: read PR main comments for more details

File: src/config/redis-info.ts [hunk 4]
```diff
@@ -6,7 +6,14 @@ export const getRedisInfo = (connectionName: string): IORedis.RedisOptions => ({
...
+		const targetError = "READONLY";
 
+		if (err.message.includes(targetError)) {
+			return 1;
```
Comment (bgvozdev) [inline src/config/redis-info.ts:13]: curious why 1 and not true?

Comment (bgvozdev) [reply src/config/redis-info.ts:13]: if true is acceptible, could be `return err.messages.includes(targetError)`

**Review by bgvozdev — APPROVED** [review]

File: src/config/redis-info.ts (hunk 4 above)
Comment (joshkay10) [reply src/config/redis-info.ts:13]: `ioredis` library expects a number, its milliseconds until reconnect, and it throws a tantrum if you pass true.

**Pull request merged**